docs/spark/how-to-guides/ubuntu-instructions.md

```diff
@@ -0,0 +1,231 @@
+       Java version: 1.8.0_191, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
+       Default locale: en, platform encoding: UTF-8
+       OS name: "linux", version: "4.4.0-17763-microsoft", arch: "amd64", family: "unix"
+       ```
+
+  4. Install **[Apache Spark 2.3+](https://spark.apache.org/downloads.html)**.
+
+     - Download [Apache Spark 2.3+](https://spark.apache.org/downloads.html) and extract it into a local folder (e.g., `~/bin/spark-2.3.2-bin-hadoop2.7`)
+     - Add the necessary [environment variables](https://www.java.com/en/download/help/path.xml) `SPARK_HOME` e.g., `~/bin/spark-2.3.2-bin-hadoop2.7/`
+
+       ```bash
+       export SPARK_HOME=~/bin/spark-2.3.2-hadoop2.7
+       export PATH="$SPARK_HOME/bin:$PATH"
+       source ~/.bashrc
+       ```
+       
+       Note that these environment variables will be lost when you close your terminal. If you want the changes to be permanent, add the `export` lines to your `~/.bashrc` file.
+
+     - Verify you are able to run `spark-shell` from your command-line        
+        Sample console output:
+        
+        ```
+        Welcome to
+              ____              __
+             / __/__  ___ _____/ /__
+            _\ \/ _ \/ _ `/ __/  '_/
+           /___/ .__/\_,_/_/ /_/\_\   version 2.3.2
+              /_/
+
+        Using Scala version 2.11.8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_201)
+        Type in expressions to have them evaluated.
+        Type :help for more information.
+
+        scala> sc
+        res0: org.apache.spark.SparkContext = org.apache.spark.SparkContext@6eaa6b0c
+        ```                      
+
+Make sure you are able to run `dotnet`, `java`, `mvn`, `spark-shell` from your command-line before you move to the next section. Feel there is a better way? Please [open an issue](https://github.com/dotnet/spark/issues) and feel free to contribute.
+
+## Build
+
+For the rest of the section, it is assumed that you have cloned the Spark .NET repository into your machine e.g., `~/dotnet.spark/`
+
+```bash
+git clone https://github.com/dotnet/spark.git ~/dotnet.spark
+```
+
+### Build Spark .NET Scala extensions layer
+
+When you submit a .NET application, .NET for Apache Spark has the necessary logic written in Scala that inform Apache Spark how to handle your requests (e.g., request to create a new Spark Session, request to transfer data from .NET side to JVM side etc.). This logic can be found in the [.NET for Apache Spark Scala Source Code](https://github.com/dotnet/spark/tree/master/src/scala).
+
+The next step is to build the .NET for Apache Spark Scala extension layer.
+
+```bash
+cd src/scala
+mvn clean package 
+```
+
+You should see JARs created for the supported Spark versions:
+
+* `microsoft-spark-2.3.x/target/microsoft-spark-2.3.x-<version>.jar`
+* `microsoft-spark-2.4.x/target/microsoft-spark-2.4.x-<version>.jar`
+
+### Build .NET sample applications using .NET Core CLI
```

Comment:
I'd add a sentence/short paragraph explaining what "sample applications" means here - introduce/give some context to what they are, link to them, etc.

Reply:
I'd also consider rewording this title (see my comment in the Windows how-to guide).